make_mock_json.py

```diff
@@ -0,0 +1,611 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import argparse
+import copy
+import logging
+import os
+import sys
+import json
+import subprocess
+from textwrap import dedent
+
+from osbs import set_logging
+from osbs.api import OSBS
+from osbs.conf import Configuration
+from osbs.constants import DEFAULT_CONFIGURATION_FILE, DEFAULT_ARRANGEMENT_VERSION
+from tests.constants import TEST_BUILD, TEST_ORCHESTRATOR_BUILD, TEST_CANCELLED_BUILD
+from osbs.exceptions import OsbsException, OsbsResponseException
+from osbs.cli.capture import setup_json_capture
+
+
+OS_VERSION = "3.9.41"
+ORCH_BUILD_LOG = dedent("""2017-06-23 17:18:41,791 platform:- - atomic_reactor.foo - DEBUG - this is from the orchestrator build
+2017-06-23 17:18:41,791 platform:x86_64 - atomic_reactor.foo - INFO - 2017-06-23 17:18:41,400 platform:- atomic_reactor.foo -  DEBUG - this is from a worker build
+2017-06-23 05:04:51,334 platform:x86_64 - atomic_reactor.plugins.orchestrate_build - INFO - "ContainersPaused": 0,
+2017-06-23 17:18:41,791 - I really like bacon
+""")
+BASE_BUILD_LOG = u"   líne 1   \n".encode('utf-8')
+MOCK_CONFIG_MAP = {
+    "apiVersion": "v1",
+    "data": {
+        "special.how": "\"very\"",
+        "special.type": "{\"quark\": \"charm\"}",
+        "config.yaml": "version:1",
+        "config.yml": "version:2",
+        "config.ymlll": "{\"version\":3}",
+        "config.json": "{\"version\":4}"
+    },
+    "kind": "ConfigMap",
+    "metadata": {
+        "creationTimestamp": "2017-06-14T16:49:47Z",
+        "name": "special-config",
+        "namespace": "osbs-qa01",
+        "resourceVersion": "117005745",
+        "selfLink": "/api/v1/namespaces/osbs-qa01/configmaps/special-config",
+        "uid": "797c8350-5121-11e7-9ce0-0efeb5d2209c"
+    }
+}
+MOCK_IMAGE_STREAM = {
+    "apiVersion": "v1",
+    "kind": "ImageStream",
+    "metadata": {
+        "annotations": {
+            "openshift.io/image.dockerRepositoryCheck": "2017-01-06T19:35:54Z",
+            "openshift.io/image.insecureRepository": "true"
+        },
+        "creationTimestamp": "2017-01-06T19:35:52Z",
+        "generation": 2,
+        "name": "test_imagestream",
+        "namespace": "test_namespace",
+        "resourceVersion": "5722991",
+        "selfLink": "/oapi/v1/namespaces/test_namespace/imagestreams/test_imagestream",
```

Comment:
Ouch, sorry we planned terribly. I'm just working on change to use OCP groupified APIs instead of `oapi/`

https://docs.openshift.com/container-platform/3.11/rest_api/apis-image.openshift.io/v1.ImageStreamTag.html

Diff:
```
"apiVersion": "image.openshift.io/v1",
...
"selfLink": "/api/image.openshift.io/v1/namespaces/test_namespace/imagestreams/test_imagestream",
```

Reply:
Also OCP4, doesn't have `oapi/` anymore :(

Reply:
@mlangsdorf is this final version? I can review it, merge and then upgrade accordingly

Reply:
I am not really happy with hardcoding in the imagestream stuff, but I cannot figure out the commands that lucarval used to generate the original imagestream jsons.

Let me clean up the lint errors that I can clean up and then you can review it again.

Reply:
Yes, as far as I'm aware ImageStreams need to be imported as JSON/YAML objects. The only image-related 'oc' commands we use are import-image and tag.